File: VB-Examples/23_Worksheets/MoveChartsheet.vb
Imports Spire.Xls
Imports System.Text

Namespace MoveChartsheet
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the workbook from a file using a relative path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\MoveChartsheet.xlsx")

            ' Move the first chartsheet in the workbook to index 2 (moves it to the third position)
            workbook.Chartsheets(0).MoveSheet(2)

            ' Move the first chartsheet in the workbook to the beginning (index 0)
            workbook.Chartsheets(0).MoveChartsheet(0)

            ' Specify the output file name for saving the modified workbook
            Dim result As String = "MoveChartSheetResult.xlsx"

            ' Save the workbook to a file with the specified output name and Excel version
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/23_Worksheets/MoveChartsheet.cs
using Spire.Xls;
using System;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace MoveChartsheet
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\MoveChartsheet.xlsx");

            //Move the first chartsheet to the position of the third sheet(including chartsheet and worksheet) 
            workbook.Chartsheets[0].MoveSheet(2);

            //Move the first sheet to the position of the first chartsheet
            workbook.Chartsheets[0].MoveChartsheet(0);

            //Save to file
            string result = "MoveChartSheetResult.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources 
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
